File: backend/app/schemas/workflow.py
from pydantic import BaseModel, Field
from typing import List, Optional
from datetime import datetime


class Approver(BaseModel):
    userId: str
    name: str
    email: str
    role: str


class WorkflowLevel(BaseModel):
    level: int
    approvers: List[Approver]


class CreateWorkflowRequest(BaseModel):
    companyId: str
    name: str
    formIds: List[str] = Field(..., min_length=1)
    levels: List[WorkflowLevel]


class UpdateWorkflowRequest(BaseModel):
    name: str
    formIds: List[str] = Field(..., min_length=1)
    levels: List[WorkflowLevel]


class WorkflowResponse(BaseModel):
    id: str
    companyId: str
    name: str
    formIds: List[str] = []
    levels: List[WorkflowLevel]
    createdBy: str
    createdAt: datetime
    updatedAt: datetime

File: frontend/src/api/workflow.ts
import API from './client';

export interface Approver {
  userId: string;
  name: string;
  email: string;
  role: string;
}

export interface WorkflowLevel {
  level: number;
  approvers: Approver[];
}

export interface WorkflowRequest {
  companyId: string;
  name: string;
  formIds: string[];
  levels: WorkflowLevel[];
}

export interface WorkflowUpdateRequest {
  name: string;
  formIds: string[];
  levels: WorkflowLevel[];
}

export interface WorkflowResponse {
  id: string;
  companyId: string;
  name: string;
  formIds: string[];
  levels: WorkflowLevel[];
  createdBy: string;
  createdAt: string;
  updatedAt: string;
}

export const workflowApi = {
  listWorkflowsByCompany: async (companyId: string): Promise<WorkflowResponse[]> => {
    const response = await API.get(`/workflows/company/${companyId}`);
    return response.data;
  },

  getWorkflow: async (workflowId: string): Promise<WorkflowResponse> => {
    const response = await API.get(`/workflows/item/${workflowId}`);
    return response.data;
  },

  createWorkflow: async (data: WorkflowRequest) => {
    const response = await API.post('/workflows', data);
    return response.data;
  },

  updateWorkflow: async (workflowId: string, data: WorkflowUpdateRequest) => {
    const response = await API.put(`/workflows/${workflowId}`, data);
    return response.data;
  },

  deleteWorkflow: async (workflowId: string) => {
    const response = await API.delete(`/workflows/${workflowId}`);
    return response.data;
  },
};
